desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java

```diff
@@ -751,7 +751,7 @@ protected boolean onWindowClosing() {
 
                 if (unsavedChangesChoice == 0) { // save changes
                     _saveButton.doClick();
-                } else if (unsavedChangesChoice == 2) { // cancel closing
+                } else if (unsavedChangesChoice != 1) { // cancel closing
```

Comment:
Yes, we have the meaning in our mind. But the actual code of JOptionPane doesn't behave according to our mind, it returns the index in the array. Not the constant. See the full code of
```
public static int showOptionDialog(Component parentComponent,
        Object message, String title, int optionType, int messageType,
        Icon icon, Object[] options, Object initialValue)
```
If the caller provides the `options` parameter, the only constant it may return is `CLOSED_OPTION` (if user closes dialog without choice). Otherwise it returns an index in the `options` array. So NOT a YES/NO constant.
Then, interpreting an array index as a constant would be dangerous. It would be mixing apples and pears.